internal/controller/rotators/aws_oidc_rotator.go

```diff
@@ -0,0 +1,199 @@
+// Copyright Envoy AI Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package rotators
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// AWSOIDCRotator implements the Rotator interface for AWS OIDC token exchange.
+// It manages the lifecycle of temporary AWS credentials obtained through OIDC token
+// exchange with AWS STS.
+type AWSOIDCRotator struct {
+	// ctx provides a user specified context.
+	ctx context.Context
```

Comment:
context is not for being stored as a field in general (it's a general anti-pattern). Just pass it as a first function argument whenever it is used. How do you know the scope of this context otherwise?